client/src/main/java/io/pravega/client/admin/impl/ReaderGroupManagerImpl.java

```diff
@@ -95,6 +96,16 @@ public void createReaderGroup(String groupName, ReaderGroupConfig config) {
                                               new ReaderGroupStateUpdatesSerializer(), new ReaderGroupStateInitSerializer(), SynchronizerConfig.builder().build());
         Map<SegmentWithRange, Long> segments = ReaderGroupImpl.getSegmentsForStreams(controller, config);
         synchronizer.initialize(new ReaderGroupState.ReaderGroupStateInit(config, segments, getEndSegmentsForStreams(config)));
+
+        if (config.isSubscriber()) {
+            Set<Stream> streams = config.getStartingStreamCuts().keySet();
+            streams.forEach(s -> getAndHandleExceptions(controller.addSubscriber(scope, s.getStreamName(), groupName)
+                            .exceptionally(e -> {
+                                System.err.println("Failed to add subscriber" + e);
+                                throw Exceptions.sneakyThrow(e);
```

Comment:
Failing to addSubscriber to Controller  should cause RG creation to fail.
Instead of throwing a RuntimeException or SneakyThrows we should throw a CheckedException as user should have a predetermined way to handle addSubscriber Failures..... perhaps retry indefinitely....

Reply:
The method on failure throws the Exception

Reply:
The retries are done on the controller end after which either a result is obtained or an exception is thrown. Using `getAndHandleExceptions`, these exceptions can be handled.

Reply:
Why do we do a "sneakyThrows" here? I would expect the application to catch this exception and do something about it.

Reply:
Please add javadoc as discussed